src/main/scala/nl/knaw/dans/easy/stage/lib/JSON.scala

```diff
@@ -99,7 +96,7 @@ object JSON {
 
     def redirectFileDatastreamJson = {
       createJSON(
-        ("dsLocation" -> dsLocation.toURI.toASCIIString) ~
+        ("dsLocation" -> settings.datastreamLocation.get.toURI.toASCIIString) ~
```

Comment:
Well, it's either datastreamLocation or file, right? So you can't have both at the same time in a Settings object? In that case that should be reflected in the types​ in Settings. I'd say an abstract class with two subclasses would do the trick. However, given the complexity of this project I'm not sure if this works. Maybe we can look into this next week in a pairprogramming session?